src/OpenSSL/SSL.py

```diff
@@ -1456,7 +1456,8 @@ def send(self, buf, flags=0):
         if not isinstance(buf, bytes):
             raise TypeError("data must be a memoryview, buffer or byte string")
 
-        result = _lib.SSL_write(self._ssl, buf, len(buf))
+        buf_len = min(len(buf), 2147483647)  # cannot send more than 2**31-1 bytes at once.
```

Comment:
This will silently truncate. I think the previous behavior of blowing up if you try to send too much here is appropriate.